provisioning/WFLY-21122-WildFly-Feature_pack-Family.adoc

```diff
@@ -0,0 +1,214 @@
+* Suppress (or at least deprecate and remove in a next version) the extra Preview feature-packs that have been defined to work with WildFly Preview: 
+cloud Preview, datasources Preview and GRPC Preview.
+* No enforcement for extra feature-packs to benefit from this feature. Already released feature-pack could be provisioned with new WildFly feature-packs
+
+NOTE: That is outside of this proposal but WildFly and WildFly Preview feature-packs, 
+in order to be valid alternatives, must expose a common set of Galleon `features`, `packages` and `layers`.
+
+=== Implementation notes
+
+The changes in the WildFly project are:
+
+* Bump to Galleon 7.x and WildFly Galleon Plugins 8.1.x
+* Update the three `wildfly-feature-pack-build.xml` files to define their family and criteria.
+* Update the WildFly full `wildfly-feature-pack-build.xml` file to inherit criteria from its WildFly EE dependency.
+
+`wildfly-feature-pack-build.xml` extract:
+
+```
+<build xmlns="urn:wildfly:feature-pack-build:3.5" producer="wildfly@maven(org.jboss.universe:community-universe):current">
+    <family name="wildfly">
+      <criteria name="microprofile"/>
+      <criteria name="microprofile-7.1"/>
+    </family>
+    <dependencies>
+        <dependency group-id="org.wildfly" artifact-id="wildfly-ee-galleon-pack" allowedFamily="wildfly:jakarta-ee-deployment+jakarta-ee">
+    ...
+</build>
+```
+
+The `allowedFamily` attribute defines the criteria that are inherited from `wildfly-ee-galleon-pack`. We can see that `jakarta-ee-11` is not inherited.
+This is a choice to not bound WildFly feature-pack to a specific version of Jakarta EE. This choice could be revisited.
+
+An example of a feature-pack that depends on WildFly EE 10:
+
+```
+<build xmlns="urn:wildfly:feature-pack-build:3.5" producer="org.example:myfeaturepack">
+    <dependencies>
+        <dependency group-id="org.wildfly" artifact-id="wildfly-ee-galleon-pack" allowedFamily="wildfly:jakarta-ee-10">
+    <dependencies>
+    ...
+</build>
+```
+
+==== Potential impact for WildFly Glow
+
+Today WildFly Glow CLI and its integration in the WildFly Maven Plugin offer the `wildfly-preview` option to contextualize the feature-packs discovery to use WildFly Preview feature-packs. This option should be deprecated
+and a new option `wildfly-variant=<variant name>` should be introduced. An immediate variant is `preview`, but new ones could be introduced in the future.
+
+=== Non-Requirements
+
+* It is not a requirement that the same set of exposed criteria is present for each new WildFly Major releases.
+
+=== Future Work
+
+None
+
+== Backwards Compatibility
+
+* This evolution is transparent and doesn't require an evolution of the provisioning tooling (WildFly Maven Plugin, WildFly Glow, ...).
+* New extra feature-packs (that make the choice to benefit from this feature) will still be compatible with older WildFly releases. Although it is advised to
```

Comment:
"will still be compatible with older WildFly releases (unless they choose to become incompatible for other reasons.)"

This is me being lawyerly. :)